drivers/pci/pcie_designware.c

```diff
@@ -22,23 +22,41 @@ static uint64_t format_dbi_addr(enum dw_pcie_access_type type, uint32_t reg)
 		break;
 	case DW_PCIE_SHADOW:
 		/* route0 = 1; */
+#ifdef CONFIG_DPU_GEN2
+		route1 = 2;
+#else
 		route1 = 1;
+#endif
 		break;
 	case DW_PCIE_ATU:
 		/* route0 = 3; */
+#ifdef CONFIG_DPU_GEN2
+		route1 = 6;
+#else
 		route1 = 2;
+#endif
 		break;
 	case DW_PCIE_DMA:
 		/* route0 = 3; */
+
+#ifdef CONFIG_DPU_GEN2
+		route1 = 7;
+#else
 		route1 = 3;
+#endif
 		break;
 	default:
 		con_err("dw_pcie: Other types is not supported\n");
 		return 0xffffffff;
 	}
 
 	/* addr = (route0 << 31) | (route1 << 17) | (uint64_t)reg; */
+
+#ifdef CONFIG_DPU_GEN2
+	addr = (route1 << 19) | reg;
+#else
 	addr = (route1 << 17) | reg;
+#endif
 	return addr;
 }
 
```

Comment:
IMO, you should get the whole format_dbi_addr function re-implemented for DPU_GEN2.
Say
```
#ifdef CONFIG_DPU_GEN2
uint64_t format_dbi_addr(enum dw_pcie_access_type type, uint32_t reg)
{
}
#else
uint64_t format_dbi_addr(enum dw_pcie_access_type type, uint32_t reg)
{
}
#endif
```